src/scripts/move.js

```diff
@@ -0,0 +1,247 @@
+'use strict';
+
+// import
+const {
+  getNotEmptyFields,
+  getNotEmptyFieldsCoords,
+  checkBetween,
+  randomField,
+} = require('./utils');
+
+const gameScore = document.querySelector('.game-score');
+const gameField = document.querySelector('.game-field');
+const allCells = gameField.querySelectorAll('.field-cell');
+const highestScore = document.querySelector('.highest-score');
+const values = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+function createItemInRandomEmptyField() {
+  const emptyFields = [];
+
+  for (const cell of allCells) {
+    if (cell.classList.length === 1) {
+      emptyFields.push(cell);
+    }
+  }
+
+  // create new item:
+  if (emptyFields.length > 0) {
+    const randomIndex = randomField(0, emptyFields.length - 1);
+    const randomValue = randomField(0, values.length - 1);
+
+    emptyFields[randomIndex]
+      .classList.add('field-cell--' + values[randomValue]);
+
+    emptyFields[randomIndex]
+      .innerText = values[randomValue];
+
+  // if its impossible to create a new item:
+  } else {
+    document.querySelector('.message-lose').classList.remove('hidden');
+    document.querySelector('.message-play').classList.add('hidden');
+
+    if (Number(localStorage.getItem('score')) < Number(gameScore.innerText)) {
+      localStorage.setItem('score', gameScore.innerText);
+
+      highestScore.innerText = localStorage.getItem('score');
+    }
+
+    document.body.removeEventListener('keyup', arrowsPlayHandler);
+  }
+}
+
+// Main movement algorithm:
+const arrowsPlayHandler = (e) => {
+  let isMoved = false;
+
+  if (e.key === 'ArrowDown') {
+    // MOVE DOWN:
+    // console.log('DOWN');
+
+    const notEmptyFields = getNotEmptyFields(allCells);
+    const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+
+    for (const field of notEmptyFields.reverse()) {
+      const fieldPos = field.id.split('-').slice(1);
+
+      for (const cell of [...allCells].reverse()) {
+        const cellPos = cell.id.split('-').slice(1);
+
+        if (fieldPos[0] < cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length === 1
+          && cell.id !== field.id
+        ) {
+          moveCell(field, cell);
+          isMoved = true;
+          break;
+        } else if (fieldPos[0] < cellPos[0]
+          && fieldPos[1] === cellPos[1]
+          && cell.classList.length !== 1
+          && cell.id !== field.id
+          && field.innerText === cell.innerText
+          && checkBetween(fieldPos, cellPos, notEmptyFieldsPositions, 0)
+        ) {
+          margeCells(field, cell);
+          isMoved = true;
+          break;
+        }
```

Comment:
we definitely can remove this `else` here

Reply:
Better to somehow move this checks to helper functions, 'cause they are too big and it's hard to read

Reply:
Also, can we make this logic reusable? I mean, seems that there is a kind of regularity in this checks and performed actions, better to separate this logic and make it reusable

Reply:
For example pass to function values that had to be compared